src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.DefaultProxyCredentials.cs

```diff
@@ -42,87 +43,90 @@ public void SetGet_Roundtrips()
         {
-            int port;
-            Task<LoopbackGetRequestHttpProxy.ProxyResult> proxyTask = LoopbackGetRequestHttpProxy.StartAsync(out port, requireAuth: true, expectCreds: true);
-            Uri proxyUrl = new Uri($"http://localhost:{port}");
-
             var rightCreds = new NetworkCredential("rightusername", "rightpassword");
             var wrongCreds = new NetworkCredential("wrongusername", "wrongpassword");
+            string expectCreds = "Basic " + Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes($"{rightCreds.UserName}:{rightCreds.Password}"));
 
-            using (HttpClientHandler handler = CreateHttpClientHandler())
-            using (var client = new HttpClient(handler))
+            await LoopbackServer.CreateServerAsync(async (proxyServer, proxyUrl) =>
             {
-                handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, rightCreds);
-                handler.DefaultProxyCredentials = wrongCreds;
-
-                Task<HttpResponseMessage> responseTask = client.GetAsync(Configuration.Http.RemoteEchoServer);
-                Task<string> responseStringTask = responseTask.ContinueWith(t =>
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
                 {
-                    using (t.Result) return t.Result.Content.ReadAsStringAsync();
-                }, TaskScheduler.Default).Unwrap();
-                await (new Task[] { proxyTask, responseTask, responseStringTask }).WhenAllOrAnyFailed();
+                    handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, rightCreds);
+                    handler.DefaultProxyCredentials = wrongCreds;
 
-                TestHelper.VerifyResponseBody(responseStringTask.Result, responseTask.Result.Content.Headers.ContentMD5, false, null);
-                Assert.Equal(Encoding.ASCII.GetString(proxyTask.Result.ResponseContent), responseStringTask.Result);
-
-                string expectedAuth = $"{rightCreds.UserName}:{rightCreds.Password}";
-                Assert.Equal(expectedAuth, proxyTask.Result.AuthenticationHeaderValue);
-            }
+                    Task<HttpResponseMessage> responseTask = client.GetAsync(Configuration.Http.RemoteEchoServer);
+                    await proxyServer.AcceptConnectionAsync(async connection =>
+                    {
+                        await connection.ReadRequestHeaderAsync().ConfigureAwait(false);
+                        if (connection.GetRequestHeaderValue("Proxy-Authorization") == null)
+                        {
+                            await connection.SendResponseAsync(HttpStatusCode.ProxyAuthenticationRequired, "Proxy-Authenticate: Basic\r\n");
+                            lines = await connection.ReadRequestHeaderAsync().ConfigureAwait(false);
+                        }
+                        // Verify that we got rightCreds instead of wrongCreds (or nothing).
```

Comment:
What does the "or nothing" part of this comment mean?

Reply:
what I meant was that this will also fail if credentials are missing. 
Maybe that is obvious.

Reply:
Ah, I understand now.  I was reading it as "verify we got rightCreds... or nothing", as if nothing was an acceptable result.

Reply:
Can we change the naming here?  "rightCreds" vs. "wrongCreds" is really confusing.  It would be better for those to be named something like "defaultSystemProxyCreds" vs. "explicitProxyCreds" etc.

Reply:
yes we can. That is what the test used before for the variable names. I can either change test to be clear it refers to variable names or I can change variable names as suggested.